aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -617,6 +647,195 @@ module aptos_framework::coin {
+        let coin = withdraw<CoinType>(owner, amount);
+        let fungible_asset = convert_to_fungible_asset(coin);
+        primary_fungible_store::deposit(signer::address_of(owner), fungible_asset);
+    }
+
+    // Non-entry version for calling in another Move module.
+    public fun convert_to_fungible_asset<CoinType>(coin: Coin<CoinType>): FungibleAsset acquires FungibleAssetMigration, CoinInfo, FungibleAssetAdmin {
+        let migration = borrow_global<FungibleAssetMigration>(@aptos_framework);
+        assert!(is_registered_fungible_asset_migration_internal<CoinType>(migration), EFUNGIBLE_ASSET_MIGRATION_COINTYPE_UNINITIALIZED);
+
+        let fungible_asset_admin_address = fungible_asset_admin_address<CoinType>();
+        let fungible_asset_admin = borrow_global_mut<FungibleAssetAdmin<CoinType>>(fungible_asset_admin_address);
+
+        let amount = value(&coin);
+        merge(&mut fungible_asset_admin.v1_locked, coin);
+        fungible_asset::mint(&fungible_asset_admin.mint_ref, amount)
+    }
+
+    // Convert an amount of FungibleAsset from an owner's primary store back to
+    // Coin. This is necessary during phase 1 of the transition as the ecosystem might
+    // not have fully supported the FungibleAsset version yet.
+    public entry fun convert_to_coin_and_store<CoinType>(owner: &signer, amount: u64) acquires FungibleAssetMigration, CoinInfo, FungibleAssetAdmin, CoinStore {
+        let migration = borrow_global<FungibleAssetMigration>(@aptos_framework);
+        let asset = *smart_table::borrow(&migration.coins_to_fungible_assets, type_info::type_of<CoinType>());
+
+        let fa = primary_fungible_store::withdraw(owner, asset, amount);
+        let coin = convert_to_coin<CoinType>(fa);
+        deposit(signer::address_of(owner), coin);
+    }
+
+    // Non-entry version for calling in another Move module.
+    public fun convert_to_coin<CoinType>(fa: FungibleAsset): Coin<CoinType> acquires FungibleAssetMigration, CoinInfo, FungibleAssetAdmin {
+        let migration = borrow_global<FungibleAssetMigration>(@aptos_framework);
+        assert!(is_registered_fungible_asset_migration_internal<CoinType>(migration), EFUNGIBLE_ASSET_MIGRATION_COINTYPE_UNINITIALIZED);
+
+        let fungible_asset_admin_address = fungible_asset_admin_address<CoinType>();
+        let fungible_asset_admin = borrow_global_mut<FungibleAssetAdmin<CoinType>>(fungible_asset_admin_address);
+
+        let amount = fungible_asset::amount(&fa);
+        fungible_asset::burn(&fungible_asset_admin.burn_ref, fa);
+        extract(&mut fungible_asset_admin.v1_locked, amount)
+    }
+
+    // Functions to withdraw a total amount of a token from the user. This will try
+    // to withdraw coins first, and then any remaining amount in FungibleAssets.
+    // This returns FungibleAsset and is meant to use by a flow that supports
+    // FungibleAsset.
+    // public fun withdraw_fungible_asset(
```

Comment:
@movekevin  this method withdraws both Coin and FungibleAsset, but Coin is not in return type - how does that work?

Reply:
what?

Reply:
The original intention is to wrap the withdrawn coins into fungible asset